### PR TITLE
Refactor c3226631

### PR DESCRIPTION
**Pull request by sdusolle**
## Description

First test scenario is finished.

https://jahia.testrail.net/index.php?/cases/view/3218086&group_by=cases:section_id&group_id=323268&group_order=asc&display_deleted_cases=0

### CODE REVIEW

File: tests/cypress/page-object/digitall.home.page.ts [hunk 1]
```diff
@@ -15,9 +21,24 @@ class DigitallHomePage extends BasePage {
...
+        this.getIframeBody()
+            .contains('div[role="row"]', 'Digitall')
+            .rightclick({ force: true })
+            .should('have.class', 'context-menu-open')
+        this.getIframeBody().find(this.elements.publishSiteinAllLang).click()
+        return workflowDashboard
+    }
+
+    flushCache(){
+        this.getIframeBody().find(this.elements.cacheButton).click().debug()
```
Comment (XiangYuHe) [inline tests/cypress/page-object/digitall.home.page.ts:38]: May I ask the usage of "debug" here?

File: tests/cypress/page-object/digitall.home.page.ts [hunk 2]
```diff
@@ -15,9 +21,24 @@ class DigitallHomePage extends BasePage {
...
             .should('have.class', 'context-menu-open')
-        this.getIframeBody().contains('span', 'Edit').click()
+        this.getIframeBody().find(this.elements.editSite).click()
         return editPage
     }
+
+    publishSite() {
+        this.getIframeBody()
+            .contains('div[role="row"]', 'Digitall')
+            .rightclick({ force: true })
```
Comment (XiangYuHe) [inline tests/cypress/page-object/digitall.home.page.ts:31]: Please don't use "force: true" this sometime cause very flaky behaviours

**Comment by XiangYuHe** [comment]
Kindly fix-up the lint errors by circleci as well (squash your commits would be greatly appreciated)

File: tests/cypress/page-object/sitemap.page.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { BasePage } from './base.page'
+import { editPage } from './edit.page'
+import { workflowDashboard } from './worflow.dashboard.page'
+
+class DigitallHomePage extends BasePage {
```
Comment (XiangYuHe) [inline tests/cypress/page-object/sitemap.page.ts:5]: naming "DigitallHomePage" ?

File: tests/cypress/page-object/digitall.home.page.ts (hunk 1 above)
Comment (sdusolle) [reply tests/cypress/page-object/digitall.home.page.ts:38]: I forgot to remove it - will be removed in the next commit

File: tests/cypress/page-object/sitemap.page.ts (hunk 3 above)
Comment (VladRadan) [reply tests/cypress/page-object/sitemap.page.ts]: removed that file and class, it wasn't used by anything

File: tests/cypress/page-object/digitall.home.page.ts (hunk 2 above)
Comment (VladRadan) [reply tests/cypress/page-object/digitall.home.page.ts]: "force:true" removed, ended up being unnecessary with the stabilization around iframes.

**Review by VladRadan — CHANGES_REQUESTED** [review]
please do not merge until https://github.com/Jahia/sitemap/pull/22 is merged

File: tests/cypress/page-object/site.home.page.ts [hunk 4]
```diff
@@ -32,9 +38,25 @@ class SiteHomePage extends BasePage {
...
+            .rightclick()
+            .should('have.class', 'context-menu-open')
+        this.getIframeBody().find(this.elements.publishSiteinAllLang).click()
+        return workflowDashboard
+    }
+
+    flushCache() {
+        this.getIframeBody().find(this.elements.cacheButton).click()
+        this.getIframeBody().find(this.elements.flushAll).click()
+        return this
```
Comment (XiangYuHe) [inline tests/cypress/page-object/site.home.page.ts:58]: Is there a check that we can do to see the flushCache is completed

File: tests/cypress/page-object/site.home.page.ts [hunk 5]
```diff
@@ -32,9 +38,25 @@ class SiteHomePage extends BasePage {
...
     }
+
+    publishSite(site = 'Digitall') {
+        this.getIframeBody()
+            .contains('div[role="row"]', site)
+            .trigger('mouseover') // Stabilize portion right before the right-click so it hover over the right element
+            .rightclick()
+            .should('have.class', 'context-menu-open')
+        this.getIframeBody().find(this.elements.publishSiteinAllLang).click()
+        return workflowDashboard
```
Comment (XiangYuHe) [inline tests/cypress/page-object/site.home.page.ts:52]: Is there a check we can do to see that publish is completed? (checking state is OK)

File: tests/cypress/page-object/site.home.page.ts [hunk 6]
```diff
@@ -1,20 +1,26 @@
...
-
-        textEdit: 'Edit',
+        publishSiteinAllLang: "[class*='publishsiteinalllanguages']",
+        editSite: "[class*='editcontentroot']",
+        cacheButton: '.edit-menu-cache',
+        flushAll: "[class*='flushall']",
     }
 
-    goTo(siteHomeUrl: string) {
+    goTo(siteHomeUrl = '/jahia/page-composer/default/en/sites/digitall/home.html') {
```
Comment (XiangYuHe) [inline tests/cypress/page-object/site.home.page.ts:17]: I would prefer URL as input for now (no default)
Can have sample in the comments

File: tests/cypress/integration/digitall.home.spec.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            .editPage('Digitall')
+            .clickOnSitemap()
+            .validateSucessMessage()
+            .clickBack()
+            .publishSite()
+            .clickPublishAll()
+            .flushCache()
+
+        // fetch the sitemap content
+        cy.request('sites/digitall/sitemap.index.xml').then((response) => {
```
Comment (XiangYuHe) [inline tests/cypress/integration/digitall.home.spec.ts:16]: Is there a way to get a XML fixture then do a comparison?
So we have a more descriptive check?

File: tests/cypress/page-object/site.home.page.ts [hunk 8]
```diff
@@ -32,9 +38,25 @@ class SiteHomePage extends BasePage {
             .rightclick()
             .should('have.class', 'context-menu-open')
 
-        this.getIframeBody().contains(this.htmlElements.span, this.elements.textEdit).click()
+        this.getIframeBody().find(this.elements.editSite).click()
```
Comment (XiangYuHe) [inline tests/cypress/page-object/site.home.page.ts:41]: May I ask why we are changing this one? (Is it causing flakiness?)

File: tests/cypress/page-object/worflow.dashboard.page.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { BasePage } from './base.page'
+import { siteHomePage } from './site.home.page'
+
+class WorkflowDashboardPage extends BasePage {
+    elements = {
+        publishAll: "[class *= 'bbar'] [class*='button-bypassworkflow']",
+    }
+
+    clickPublishAll() {
+        this.getIframeBody().find(this.elements.publishAll).click()
```
Comment (XiangYuHe) [inline tests/cypress/page-object/worflow.dashboard.page.ts:10]: Do we want to check if there is something to publish?
Or just hard publish without going through the workflow dashboard?

File: tests/cypress/page-object/site.home.page.ts (hunk 8 above)
Comment (VladRadan) [reply tests/cypress/page-object/site.home.page.ts:41]: it was changed to accommodate a case where the UI isn't in English

File: tests/cypress/integration/digitall.home.spec.ts (hunk 7 above)
Comment (VladRadan) [reply tests/cypress/integration/digitall.home.spec.ts:16]: not familiar with such tools or libraries

File: tests/cypress/page-object/worflow.dashboard.page.ts (hunk 9 above)
Comment (VladRadan) [reply tests/cypress/page-object/worflow.dashboard.page.ts:10]: the purpose is not to test the workflow, it's just to publish the site to see that sitemap is working

File: tests/cypress/page-object/site.home.page.ts (hunk 6 above)
Comment (VladRadan) [reply tests/cypress/page-object/site.home.page.ts]: sure, works either way for me

File: tests/cypress/page-object/site.home.page.ts (hunk 5 above)
Comment (VladRadan) [reply tests/cypress/page-object/site.home.page.ts:52]: added

File: tests/cypress/page-object/site.home.page.ts (hunk 4 above)
Comment (VladRadan) [reply tests/cypress/page-object/site.home.page.ts:60]: added

**Review by VladRadan — APPROVED** [review]

File: tests/cypress/page-object/site.home.page.ts [hunk 10]
```diff
@@ -32,9 +38,27 @@ class SiteHomePage extends BasePage {
...
         return editPage
     }
+
+    publishSite(site = 'Digitall') {
```
Comment (XiangYuHe) [inline tests/cypress/page-object/site.home.page.ts:45]: Probably move this to input parameter

**Review by XiangYuHe — APPROVED** [review]

File: tests/cypress/page-object/site.home.page.ts (hunk 10 above)
Comment (VladRadan) [reply tests/cypress/page-object/site.home.page.ts]: done

**Pull request merged**